docs/pages/developers/evm/filler.mdx

```diff
@@ -1,272 +1,796 @@
+
+class MyArbitrageStrategy implements FillerStrategy {
+	name = "ArbitrageStrategy"
+
+	async canFill(order: Order): Promise<boolean> {
+		// Check if arbitrage opportunity exists
+		const priceA = await this.getTokenPrice(order.inputs[0].token, order.sourceChain)
+		const priceB = await this.getTokenPrice(order.outputs[0].token, order.destChain)
+		return priceA < priceB * 0.95 // 5% arbitrage threshold
+	}
+
+	async calculateProfitability(order: Order): Promise<bigint> {
+		// Calculate arbitrage profit + order fees - costs
+		const arbitrageProfit = await this.calculateArbitrage(order)
+		const orderFees = order.fees
+		const costs = await this.estimateCosts(order)
+		return arbitrageProfit + orderFees - costs
+	}
+
+	async executeOrder(order: Order): Promise<ExecutionResult> {
+		// Execute complex arbitrage + fill strategy
+		return await this.executeArbitrageAndFill(order)
+	}
+}
+```
+
+## Configuration
+
+### Confirmation Policy - Security vs Speed
+
+The confirmation policy is a **critical security feature** that determines how many blockchain confirmations to wait for before processing an order. This prevents attacks from blockchain reorganizations (reorgs) where transactions can be reversed.
+
+#### How It Works
+
+The filler uses a **sliding scale** based on order value:
+
+- **Low-value orders**: Fewer confirmations (faster processing)
+- **High-value orders**: More confirmations (higher security)
+
+```
+Confirmations = Linear interpolation between min/max based on order value
+```
+
+#### Configuration Format
+
+```toml
+[confirmationPolicies."1"]  # Ethereum mainnet
+minAmount = "1000"      # 1000 USD
+maxAmount = "10000"    # 10000 USD
+minConfirmations = 12                  # 12 blocks for small orders (~3 minutes)
+maxConfirmations = 30                  # 30 blocks for large orders (~7.5 minutes)
+
+[confirmationPolicies."97"]  # BSC testnet
+minAmount = "1000"      # 1000 USD
+maxAmount = "10000"    # 10000 USD
+minConfirmations = 1                   # 1 block for small orders (~3 seconds)
+maxConfirmations = 5                   # 5 blocks for large orders (~15 seconds)
+```
+
+#### Example Calculation
+
+For a 5000 USD order on Ethereum:
+
+- Min: 1000 USD → 12 confirmations
+- Max: 10000 USD → 30 confirmations
+- Order: 5000 USD → 20 confirmations
+
+#### Chain-Specific Recommendations (With Approx Block Time Values)
+
+| Chain    | Block Time | Low-Value | High-Value | Reasoning                                   |
+| -------- | ---------- | --------- | ---------- | ------------------------------------------- |
+| Ethereum | ~12s       | 12 blocks | 30 blocks  | Highest security, slower finality (~6 min) |
+| Arbitrum | ~0.25s     | 20 blocks | 100 blocks | Optimistic rollup, 7-day challenge period  |
+| Base     | ~2s        | 10 blocks | 50 blocks  | Optimistic rollup on Ethereum L2           |
+| BSC      | ~1s        | 5 blocks  | 20 blocks  | Faster blocks, validator-based consensus    |
```

Comment:
```suggestion
| Ethereum | ~12s       | 2 blocks | 20 blocks  | Highest security, slower finality (~6 min) |
| Arbitrum | ~0.25s     | 20 blocks | 100 blocks | Optimistic rollup, 7-day challenge period  |
| Base     | ~2s        | 10 blocks | 50 blocks  | Optimistic rollup on Ethereum L2           |
| BSC      | ~1s        | 5 blocks  | 10 blocks  | Faster blocks, validator-based consensus    |
```

I'd advocate for lower number of blocks for ethereum & bsc. Ethereum rarely re-orgs, bsc has instant finality technically